fabfile/run.py

```diff
@@ -203,3 +202,31 @@ def tpch_queries(query_info, connectionURI, pg_version, citus_version, config_fi
         out_val = run(run_string)
         results_file.write(out_val)
         results_file.write('\n')
+
+@task
+@roles('master')
+def valgrind(*args): 
+    'Runs valgrind tests'
+
+    citus_repo = config.settings['community-enterprise']
+
+    # install citus and set citus path variable
+    if citus_repo == 'community':
+        repo_path = config.CITUS_REPO
+    elif citus_repo == 'enterprise':
+        repo_path = config.ENTERPRISE_REPO
+    
+    setup.valgrind()
+
+    with cd(os.path.join(repo_path, config.RELATIVE_REGRESS_PATH)):
+        run('make check-multi-vg VALGRIND_LOG_FILE=logs.txt')
+        
+        # ship output files to result file in order to push to github
```

Comment:
For better readability you can have a method that filters the valgrind logs and a method that moves the outputs. So the structure could be like this:

```python
filter_valgrind_logs()

move_valgrind_outputs()
```

However I have one question, instead of moving the outputs, why dont we directly generate them in the desired output directory(results)?

Reply:
done